### DISCUSSION TITLE
How to integrate multi-dimensional edge features to GCN?

### DISCUSSION THREAD

**Discussion by Shiiivaa** [pyg-team/pytorch_geometric#7229] [Q&A]
Hello,
Currently, I am using a combination of GCN and GAT layers for a node regression task in my GNN structure. This setup works well when using only node features. However, I now want to incorporate multi-dimensional edge features into my GNN structure.
The problem is that GCN seems to only accept 1-dimensional features (edge weights). Can you help me figure out **how to integrate multi-dimensional edge features into GCN while still maintaining the current structure** (i.e., combination of GCN and GAT layers)? I would prefer not to switch to a different GNN structure.

**Comment by wsad1** [comment]
1. I think its best to not use a `GCN` if you want to use multi-dimensional edge features. Instead you could use any layer that supports `edge_attr` from [here](https://pytorch-geometric.readthedocs.io/en/latest/cheatsheet/gnn_cheatsheet.html), like `GATConv`.
2. If you really want to stick with `GCN` you could apply a linear transform to your `edge_attr` to transform it to a 1-d tensor and pass it as `edge_wieght` to `GCN`.

**Comment by riyajatar37003** [comment]
so let's assume if i have edge feature are in 2dim. u mean to say first project them into 1-d and use them as edge weight.?
should i keep this linear transform layer as learnable.?
